lib/membrane_rtc_engine/endpoints/webrtc/variant_tracker.ex

```diff
@@ -64,23 +89,24 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.VariantTracker do
   """
   @spec reset(t()) :: t()
   def reset(tracker) do
-    %__MODULE__{tracker | samples: 0, cycles: 0}
+    %__MODULE__{tracker | samples: 0, activity_cycles: 0}
   end
 
   defp maybe_inactive(tracker) do
-    if tracker.status == :active do
+    if tracker.status == :active and
+         tracker.inactivity_cycles == tracker.required_inactivity_cycles do
       Membrane.Logger.debug("Variant #{inspect(tracker.variant)} is inactive.")
-      tracker = %__MODULE__{tracker | status: :inactive}
+      tracker = %__MODULE__{tracker | status: :inactive, samples: 0}
```

Comment:
```suggestion
      tracker = %__MODULE__{tracker | status: :inactive, inactivity_cycles: 0}
```

Reply:
If we don't clear samples, this could break activity detection in the case where `required_activity_cycles` is 1.